core/src/main/java/org/elasticsearch/action/support/nodes/BaseNodesResponse.java

```diff
@@ -91,11 +105,36 @@ public TNodeResponse getAt(int position) {
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         clusterName = ClusterName.readClusterName(in);
+        failures = in.readArray(FailedNodeException[]::new, FailedNodeException::new);
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         clusterName.writeTo(out);
+        out.writeArray(failures);
     }
+
+    /**
+     * Expected to be invoked by any {@code BaseNodesResponse} that implements {@code ToXContent} at the top level of the response object.
+     */
+    protected XContentBuilder toInnerXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
> I'm going to add the static method, but I do want to note that the method name is toInnerXContent rather than toXContent, so it's not overridden by any of the child implementations.

Sure but it *can* be overridden, if it is overridden it must be called, and it has to be called by the `toXContent` methods on the inheriting classes that do implement `ToXContent` The typical pattern to address this is to make `toXContent` final and have it delegate to an abstract `doToXContent` inner method that the inheriting classes must override. But the reason that I do not like that solution here is because not all of the inheriting classes will implement `toXContent` so I do not think this method should be on the super class at all.